calculator/src/main/java/com/market/banica/calculator/service/JMXServiceImpl.java

```diff
@@ -0,0 +1,299 @@
+package com.market.banica.calculator.service;
+
+import com.market.banica.calculator.data.contract.RecipesBase;
+import com.market.banica.calculator.enums.UnitOfMeasure;
+import com.market.banica.calculator.model.Product;
+import com.market.banica.calculator.service.contract.JMXService;
+import com.market.banica.calculator.service.contract.ProductService;
+import io.micrometer.core.lang.Nullable;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+
+@EnableMBeanExport
+@ManagedResource
+@Service
+@RequiredArgsConstructor
+public class JMXServiceImpl implements JMXService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JMXServiceImpl.class);
+
+    private final RecipesBase recipesBase;
+    private final ProductService productService;
+
+    @Override
+    @ManagedOperation
+    public Map<String,Product> getDatabase(){
+      return   recipesBase.getDatabase();
+    }
+
+    @Override
+    @ManagedOperation
+    public void createProduct(String newProductName, String unitOfMeasure, Map<String, Integer> ingredients) {
+        LOGGER.debug("JMX server impl: In createProduct method");
+        LOGGER.info("CreateProduct called from JMX server with parameters newProductName {},unitOfMeasure {}," +
+                        " ingredients {} and ingredients {}", newProductName, unitOfMeasure,
+                ingredients, ingredients);
+        Product product = retrieveProductFromDatabase(newProductName);
+
+        if (product == null) {
```

Comment:
If we expect in some cases to have a null value please consider Optional

Reply:
Done